src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -180,6 +180,7 @@ def load_arguments(self, _):
         c.argument('addons', options_list=['--addons', '-a'], validator=validate_addons)
         c.argument('subnet_name', options_list=['--subnet-name', '-s'])
         c.argument('disable_sgxquotehelper', action='store_true')
+        c.argument('osm_mesh_name', options_list=['--osm-mesh-name'])
```

Comment:
although we validate the mesh-name in the server side, it's better to have the same validation on the client side.

Reply:
Considering this argument is optional and just a string, does backend validation not suffice? This property gets applied as a value for a kubernetes label.

Reply:
It is used in the label and webhook configuration name. They all need to satisfy the DNS naming validation

Reply:
Yes, but doesn't backend validation suffice?

Reply:
@robbiezhang we discussed and this should be ok as is at the moment.